plugin/selection_range.py

```diff
@@ -0,0 +1,55 @@
+from .core.protocol import Range
+from .core.protocol import Request
+from .core.registry import get_position
+from .core.registry import LspTextCommand
+from .core.sessions import method_to_capability
+from .core.typing import Any, Dict, Optional, List
+from .core.views import range_to_region
+from .core.views import selection_range_params
+import sublime
+
+
+class LspExpandSelectionCommand(LspTextCommand):
+    method = 'textDocument/selectionRange'
+    capability = method_to_capability(method)[0]
+
+    def __init__(self, view: sublime.View) -> None:
+        super().__init__(view)
+        self._regions = []  # type: List[sublime.Region]
+        self._change_count = 0
+
+    def run(self, edit: sublime.Edit, event: Optional[dict] = None) -> None:
+        session = self.session(self.capability, get_position(self.view, event))
+        if session:
+            params = selection_range_params(self.view)
+            self._regions.extend(self.view.sel())
+            self._change_count = self.view.change_count()
+            session.send_request(Request(self.method, params), self.on_result, self.on_error)
```

Comment:
We need to trigger original key binding if there is no session for position. Otherwise expanding will just be broken when there is a session but it doesn't support expanding.